riotfile.py

```diff
@@ -287,6 +287,80 @@ def select_pys(min_version=min(SUPPORTED_PYTHON_VERSIONS), max_version=max(SUPPO
                 Venv(pys=select_pys(), pkgs={"elasticsearch7": ["~=7.6.0", "~=7.8.0", "~=7.10.0", latest]}),
             ],
         ),
+        Venv(
+            name="flask",
+            command="pytest {cmdargs} tests/contrib/flask",
+            pkgs={
+                "blinker": latest,
+                "pytest": "~=3.0",
```

Comment:
Is this actually required?

Reply:
Good question... just ported what we had.

Reply:
short answer... yes...

```
____ FlaskCacheWrapperTest.test_redis_cache_tracing_with_a_wrong_connection ____

self = <tests.contrib.flask_cache.test_wrapper_safety.FlaskCacheWrapperTest testMethod=test_redis_cache_tracing_with_a_wrong_connection>

    def test_redis_cache_tracing_with_a_wrong_connection(self):
        # create the TracedCache instance for a Flask app
        Cache = get_traced_cache(self.tracer, service=self.SERVICE)
>       app = Flask(__name__)

tests/contrib/flask_cache/test_wrapper_safety.py:133: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.riot/.venv_py390/lib/python3.9/site-packages/flask/app.py:357: in __init__
    instance_path = self.auto_find_instance_path()
.riot/.venv_py390/lib/python3.9/site-packages/flask/app.py:662: in auto_find_instance_path
    prefix, package_path = find_package(self.import_name)
.riot/.venv_py390/lib/python3.9/site-packages/flask/helpers.py:784: in find_package
    if _matching_loader_thinks_module_is_package(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

loader = <_pytest.assertion.rewrite.AssertionRewritingHook object at 0x7ffa2d7fe550>
mod_name = 'tests'

    def _matching_loader_thinks_module_is_package(loader, mod_name):
        """Given the loader that loaded a module and the module this function
        attempts to figure out if the given module is actually a package.
        """
        # If the loader can tell us if something is a package, we can
        # directly ask the loader.
        if hasattr(loader, 'is_package'):
            return loader.is_package(mod_name)
        # importlib's namespace loaders do not have this functionality but
        # all the modules it loads are packages, so we can take advantage of
        # this information.
        elif (loader.__class__.__module__ == '_frozen_importlib' and
              loader.__class__.__name__ == 'NamespaceLoader'):
            return True
        # Otherwise we need to fail with an error that explains what went
        # wrong.
>       raise AttributeError(
            ('%s.is_package() method is missing but is required by Flask of '
             'PEP 302 import hooks.  If you do not use import hooks and '
             'you encounter this error please file a bug against Flask.') %
            loader.__class__.__name__)
E       AttributeError: AssertionRewritingHook.is_package() method is missing but is required by Flask of PEP 302 import hooks.  If you do not use import hooks and you encounter this error please file a bug against Flask.
```

It looks like this happens on Flask < 1.0.0

Reply:
:+1: just wanted to make sure it wasn't a tox-specific thing

Reply:
This latest change appears to have fixed it, good catch though, since we don't need to pin pytest on newer versions.